sklearn/model_selection/_search.py

```diff
@@ -1251,22 +1262,26 @@ class RandomizedSearchCV(BaseSearchCV):
         Refer :ref:`User Guide <cross_validation>` for the various
         cross-validation strategies that can be used here.
 
-    refit : boolean, or string default=True
+    refit : boolean, string, or callable, default=True
         Refit an estimator using the best found parameters on the whole
         dataset.
 
         For multiple metric evaluation, this needs to be a string denoting the
         scorer that would be used to find the best parameters for refitting
         the estimator at the end.
 
+        Where there are considerations other than maximum score in
+        choosing a best estimator, ``refit`` can be set to a function which 
+        returns the selected ``best_index_`` given the ``cv_results``.
+
         The refitted estimator is made available at the ``best_estimator_``
         attribute and permits using ``predict`` directly on this
         ``RandomizedSearchCV`` instance.
 
         Also for multiple metric evaluation, the attributes ``best_index_``,
         ``best_score_`` and ``best_parameters_`` will only be available if
         ``refit`` is set and all of them will be determined w.r.t this specific
-        scorer.
+        scorer. When refit is callable, ``best_score_`` is disabled.
```

Comment:
I assume that this is because `best_score_` is not necessarily the highest (or lowest) score anymore? Isn't it still useful to report the score that is associated with the best hyperparmeter setting?

Reply:
Yes, `best_score_` may not necessarily be the highest (or lowest) score anymore, and one could always retrive this infomation once you have `best_index_`. For example, `best_score_ = cv_results_["mean_test_xxx"][best_index_]`. 

Another reason why we disable it is `refit_metric` is needed to compute `self.best_score_` as shown [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_search.py#L764) in the original code base. Our callable won't be able to tell the value of `refit_metric` in implementation. So @jnothman and I agreed to simply disable it.